### PR TITLE
feat: allow query on tiles

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

There's no way to add these except by editing the DB directly. So it'd be safe to merge as is... but also opening now to get early feedback :)

stacked on top of #13976 
see #13927 

## Changes

* Teaches the `InsightCard` how to display a QueryNode.
* is definitely not finished

dash-q-tiles

## How did you test this code?

editing the DB so that a dashboard tile points at an insight with a datatable insight and then using the UI

### CODE REVIEW

File: frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx [hunk 1]
```diff
@@ -394,6 +407,36 @@ function InsightMeta({
...
+        insightType = QUERY_TYPES_METADATA[query.kind]
+    } else if (!!filters.insight) {
+        insightType = INSIGHT_TYPES_METADATA[filters.insight]
+    } else {
+        // maintain the existing default
+        insightType = INSIGHT_TYPES_METADATA[InsightType.TRENDS]
+    }
+
+    let { date_from, date_to } = filters
+    if (!!query && isInsightQueryNode(query)) {
```
Comment (pauldambra) [inline frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx:426]: This is a little yucky...

Probably need a helper function to get the date range off any query...

Comment (thmsobrmlr) [reply frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx:426]: Yes, let's try to unify the schema for all queries with a start and end date as good as possible.

Comment (pauldambra) [reply frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx]: Pulled the calculation out of Insight Card and into schema utils for now.

**Review by mariusandra — APPROVED** [review]
Let's go!

**Pull request merged**